packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -319,9 +322,32 @@ function useId(): string {
 
 // useMemoCache is an implementation detail of Forget's memoization
 // it should not be called directly in user-generated code
-// we keep it as a stub for dispatcher
 function useMemoCache(size: number): Array<any> {
-  return [];
+  const fiber = currentFiber;
+  if (fiber == null) {
+    throw new Error(
+      'Attempted to call useMemoCache, but current Fiber ' +
+        'is not initialized. This is a bug in React DevTools.',
+    );
+  }
+
+  const memoCache = fiber.updateQueue?.memoCache;
+  if (memoCache == null) {
+    return [];
```

Comment:
Wouldn't this case need to write the cache?

Where is this implementation coming from?

Reply:
Compared to https://github.com/facebook/react/blob/ce2bc58a9f6f3b0bfc8c738a0d8e2a5f3a332ff5/packages/react-reconciler/src/ReactFiberHooks.js#L1113 it generally seems to be doing a lot less. Could this just call the "real implementation", esp if it doesn't need to do any extra record keeping?

Reply:
i think the main difference is that it's using syntax sugar which the real version has to avoid because we prefer to avoid the bloated expansion of optional chains and such. this looks right at least, though +1 to just reusing the same implementation

Reply:
It should not do the same thing as the original implementation. Other hook implementations in this file actually don't mutate anything in `Fiber` or `Hook` object.

This can't be applied for `useMemoCache` and the main reason is its implementation: although it is a hook, it is not implemented in the same way as others. It doesn't update `memoizedState` on the Fiber, so we could see it in the chain with other hooks.

When `inspectHooksOfFiber` is called, it gets a Fiber as an argument, which is already _"rendered"_, so we could traverse through `Fiber.memoizedState` and return values from there. For some reason, this doesn't work with `useMemoCache`, I've tried returning just `memoCache.data[memoCache.index]`, but this implementation produces errors in the render function of the inspected component. I am guessing that it is somewhat related to `useMemoCache`'s implementation.